doc/classes/Array.xml

```diff
@@ -250,53 +259,57 @@
 			<return type="Array" />
 			<param index="0" name="deep" type="bool" default="false" />
 			<description>
-				Returns a copy of the array.
-				If [param deep] is [code]true[/code], a deep copy is performed: all nested arrays and dictionaries are duplicated and will not be shared with the original array. If [code]false[/code], a shallow copy is made and references to the original nested arrays and dictionaries are kept, so that modifying a sub-array or dictionary in the copy will also impact those referenced in the source array. Note that any [Object]-derived elements will be shallow copied regardless of the [param deep] setting.
+				Returns a new copy of the array.
+				By default, a [b]shallow[/b] copy is returned: all nested [Array] and [Dictionary] elements are shared with the original array. Modifying them in one array will also affect them in the other.[br]If [param deep] is [code]true[/code], a [b]deep[/b] copy is returned: all nested arrays and dictionaries are also duplicated (recursively).
```

Comment:
There was some feedback about the description of the `duplicate` method in https://github.com/godotengine/godot/pull/71786#discussion_r1085737667.

@RedworkDE Do you think the description is better now?

Reply:
I'd say it better, but the (incorrect) note about objects got lost, instead of corrected.
Also we may want to elaborate on the meaning of shared, unless that is clarified elsewhere already.
Additionally the behavior when the original array contains shared elements may be unexpected.
I left suggestions on how these things could be worded but that probably needs a second (and third pass)
One last thing the recursion depth of deep copies is also limited, but I am not sure if and how this should be documented here.
```suggestion
				Returns a new copy of the array.
				By default, a [b]shallow[/b] copy is returned: all nested [Array] and [Dictionary] elements are shared with the original array. Modifying them in one array will also affect them in the other.[br]If [param deep] is [code]true[/code], a [b]deep[/b] copy is returned: all nested arrays and dictionaries are also duplicated (recursively).
				[b]Note:[/b] [Object]-derived elements are always shared with the original array regardless of the [param deep] setting.
				[b]Note:[/b] An element is shared iff a modification to one version of it is also visible to the other.
				[b]Note:[/b] If the original contained shared elements, after a deep copy, these elements will no longer be shared and instead be distinct copies.
```

Reply:
I want to note that it's worth being really careful with words, and how much to really expose to the documentation. In my past contributions to the docs I have always avoided to express way too much detail. It is quite overwhelming to any user, and they make the subject look more complex than it truly is. Oftentimes these concepts are better explained somewhere else, and merely reminded of where they matter.